libfsntfs/libfsntfs_mft_entry.c

```diff
@@ -1497,6 +1497,22 @@ int libfsntfs_mft_entry_read_attribute_list_data_mft_entry_by_index(
 	{
 		return( 0 );
 	}
+	if( (data_mft_entry->header->flags & LIBFSNTFS_MFT_ENTRY_FLAG_IN_USE) == 0 )
```

Comment:
I think they do have a generic function (`libfsntfs_mft_entry_is_allocated`) for this, but maybe I'm not paying enough attention to details.